drivers/hmis/app/graphql/types/hmis_schema/query_type.rb

```diff
@@ -407,25 +411,31 @@ def external_form_definition(identifier:)
       argument :identifier, String, required: true
     end
     def form_identifier(identifier:)
-      raise 'Access denied' unless current_user.can_configure_data_collection?
+      access_denied! unless current_user.can_configure_data_collection?
+
+      definition = Hmis::Form::Definition.non_static.latest_versions.where(identifier: identifier).first
+      raise 'not found' unless definition
+
+      ensure_form_role_permission(definition.role)
 
-      Hmis::Form::Definition.non_static.latest_versions.where(identifier: identifier).first
+      definition
     end
 
     field :form_identifiers, Types::Forms::FormIdentifier.page_type, null: false do
       filters_argument Forms::FormIdentifier
     end
     def form_identifiers(filters: nil)
-      raise 'Access denied' unless current_user.can_configure_data_collection?
+      access_denied! unless current_user.can_configure_data_collection?
 
       scope = Hmis::Form::Definition.non_static.valid.latest_versions
+      scope = scope.with_role(Hmis::Form::Definition::NON_ADMIN_FORM_ROLES) unless current_user.can_administrate_config?
       scope = scope.apply_filters(filters) if filters
       scope.order(updated_at: :desc)
     end
 
     form_rules_field
     def form_rules(**args)
-      raise 'Access denied' unless current_user.can_configure_data_collection?
+      access_denied! unless current_user.can_configure_data_collection?
```

Comment:
I still need to update this to filter out non-admin form roles if the current user lacks `can_administrate_config` permission. It should be intersected with the `form_type` filter argument if it's passed.

Reply:
Ok, I've taken a stab at this, which I think is possibly not ideal because it's duplicating logic in the [FormInstanceFilter](https://github.com/greenriver/hmis-warehouse/blob/4c4a9e76d153de126451670502e2db9b30a60b10/drivers/hmis/app/models/hmis/filter/form_instance_filter.rb#L22-L28). This is not just bad because of the duplicated logic, but also because of the double query for form definition identifiers if the user passes in additional form roles to filter by.

However, the code is potentially clearer this way, and to be honest, I wasn't easily able to figure out how to overwrite the filters object with additional filters to pass (maybe using a [preprocessing step](https://graphql-ruby.org/fields/arguments.html#preprocessing) it would be possible?)

Reply:
This looks good to me @martha . Instance has the `with_role` scope too, so we could use that in the same way you did for the form identifiers query 
```
scope = Hmis::Form::Instance.all
scope = scope.with_role(Hmis::Form::Definition::NON_ADMIN_FORM_ROLES) unless current_user.can_administrate_config?
```

Reply:
Sounds good! This needed a small tweak [here](https://github.com/greenriver/hmis-warehouse/pull/4531/files#diff-209b289761c1bb46bd55cd35081e14b1912ec5704296c27b7fcca123ef8d41f6R41) to accept a list of roles instead of a single role